dockhub.py

```diff
@@ -1,226 +1,182 @@
 #!/usr/bin/env python3
 
+from functools import wraps
 import json
 from os import getenv
-from sys import exit
+import sys
 
 import click
 import requests
 
 
-def die(msg=None, exception=None):
-    if msg is not None:
-        print(f'{msg}')
-    if exception is not None:
-        print(f'{exception}')
-    exit(1)
+CONTENT_HEADER = {'Content-Type': 'application/json', 'charset': 'utf-8'}
+DOCKERHUB_URL = 'https://hub.docker.com/v2'
 
 
+def handle_http_errors(fun):
+    """Wrap function with exception handling for common HTTP errors."""
+    @wraps(fun)
+    def _handle_http_errors(*args, **kwargs):
+        try:
+            return fun(*args, **kwargs)
+
+        except requests.ConnectionError as exc:
+            raise click.ClickException(f'Unable to connect to dockerhub: {exc}')
+        except requests.HTTPError as exc:
+            raise click.ClickException(f'HTTPError: {exc}')
+        except requests.Timeout as exc:
+            raise click.ClickException(f'Timeout: {exc}')
+        except requests.TooManyRedirects as exc:
+            raise click.ClickException(f'Too many redirects encountered: {exc}')
+
+    return _handle_http_errors
+
+
+def die(msg):
+    """Print a message and exit with 1 status code."""
+    click.echo(click.style(msg, fg="red"))
```

Comment:
click can do styling. One really nice thing about click is that it will quell the styling in a non-tty context so it doesn't hose piping or terminals that don't support ANSI character codes or anything like that.